src/librustc_privacy/lib.rs

```diff
@@ -939,8 +964,8 @@ impl<'a, 'tcx: 'a> TypeVisitor<'tcx> for SearchInterfaceForPrivateItemsVisitor<'
                     // free type aliases, but this isn't done yet.
                     return false;
                 }
-
```

Comment:
The substitutions are visited by `super_visit_with` below. *However*, see my comment on `impl_trait_ref`, you forgot to copy one line to it.